backend/src/main/java/ca/gov/dtsstn/cdcp/api/dto/Subscription.java

```diff
@@ -0,0 +1,39 @@
+package ca.gov.dtsstn.cdcp.api.dto;
+
+import java.io.Serializable;
+
+import org.immutables.value.Value;
+import org.immutables.value.Value.Immutable;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+import org.springframework.lang.Nullable;
+
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableSubscription.class)
+@JsonDeserialize(as = ImmutableSubscription.class)
+public abstract class Subscription implements Serializable {
```

Comment:
**Nitpick:** this can be an interface. There is almost no reason (that I can think of) for immutables to use abstract classes in Java 21.

Reply:
fixed